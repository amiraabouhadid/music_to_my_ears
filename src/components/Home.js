/* eslint-disable react/prop-types */
import React from 'react';
import { AiOutlineSearch } from 'react-icons/ai';
import Album from './Album';

const Home = ({ albums }) => (
  <>
    <div className="container-fluid">
      <div className="row header-row pt-2 text-white">
        <div className="col-4">
          <p className="fw-bold">2022</p>
        </div>
        <div className="col-4 text-center"><p> Top Albums</p></div>
        <div className="col-4 ">
          <AiOutlineSearch className=" fw-bold float-end" />
        </div>
      </div>
    </div>
    <div className="conatiner-fluid pink-bg">
      <div className="row g-0">
        <div className="col-6">

          {albums.slice(0, albums.length / 2).map((a, index) => (
            <Album key={a.id} album={a} index={index} columnNo="1" />
          ))}
        </div>
        <div className="col-6">

          {albums.slice(albums.length / 2).map((a, index) => (
            <Album key={a.id} album={a} index={index} columnNo="2" />
          ))}
        </div>
      </div>

    </div>
  </>
);
export default Home;
