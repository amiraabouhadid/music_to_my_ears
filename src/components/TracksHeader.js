import React from 'react';
import PropTypes from 'prop-types';

const TracksHeader = ({ album }) => (
  <>
    <div className="p-3 d-flex flex-row align-items-baseline justify-content-between">
      <div className="">
        <img src={album.imageUrl} alt={album.name} style={{ width: '100%' }} />
      </div>
      <div className="text-white text-end">
        <h2 className="">
          {album.name}
        </h2>
        <small>
          {`${album.trackCount} tracks`}
        </small>

      </div>
    </div>
  </>
);
TracksHeader.propTypes = {
  album:
    PropTypes.instanceOf(Object).isRequired,
};
export default TracksHeader;
