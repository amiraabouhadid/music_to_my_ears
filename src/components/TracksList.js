/* eslint-disable react/prop-types */
import { Spinner } from 'react-bootstrap';
import Track from './Track';

const TracksList = ({ tracks, loadingStatus }) => (
  <>
    {loadingStatus === 'loading' ? (
      <div className="py-5">
        <div className="text-light text-center py-5 px-3">
          <Spinner animation="grow" />
          <h6>
            {'Fetching all the tracks...'.toUpperCase()}
          </h6>
        </div>
      </div>
    ) : (tracks.map((track, idx) => (
      <Track idx={idx} key={track.id} track={track} />
    )))}
  </>
);

export default TracksList;
