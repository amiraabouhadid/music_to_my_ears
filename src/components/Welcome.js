import { Link } from 'react-router-dom';
import { AiFillCopyrightCircle } from 'react-icons/ai';
import { FaMusic } from 'react-icons/fa';
import { GiMusicalScore, GiMusicalNotes } from 'react-icons/gi';

const Welcome = () => {
  const links = [
    {
      id: 1,
      text: 'top albums',
      path: '/music_to_my_ears/albums/top',
      color: 'rgb(35,181,78)',
    },
    {
      id: 2,
      text: 'new albums',
      path: '/music_to_my_ears/albums/new',
      color: 'rgb(247,90,46)',
    },
    {
      id: 3,
      text: 'staff picks',
      path: '/music_to_my_ears/albums/picks',
      color: 'rgb(93, 93, 235)',
    },
  ];
  return (
    <div className="welcome-page text-light">
      <div className="p-4 dark-bg text-center">
        <FaMusic />
        <GiMusicalScore />
        <GiMusicalNotes />
        <h1>
          {'Music to my ears'.toUpperCase()}
        </h1>
      </div>
      <div className="p-5 text-center">
        <div>
          <h2>
            {'Pick a category!'.toUpperCase()}
          </h2>
        </div>

        {links.map((link) => (
          <div key={link.id} style={{ backgroundColor: `${link.color}` }} className=" border my-4 p-5">
            <h3>
              <Link className="text-light" to={link.path} style={{ textDecoration: 'none' }}>
                {link.text.toUpperCase()}
              </Link>

            </h3>
          </div>

        ))}
      </div>
      <div className="dark-bg fixed-bottom p-3 text-center">
        {'Amira Abouhadid '}
        <AiFillCopyrightCircle />
        {' Creative Commons'}

      </div>

    </div>
  );
};
export default Welcome;
