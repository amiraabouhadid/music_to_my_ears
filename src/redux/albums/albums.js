/* eslint-disable no-case-declarations */
import AlbumService from '../../services/AlbumService';

const TOP_ALBUMS_RETRIEVED = 'app/albums/TOP_ALBUMS_RETRIEVED';
const ALBUM_IMAGE_RETRIEVED = 'app/albums/ALBUMS_IMAGE_RETRIEVED';
const reducer = (state = [], action) => {
  const { type, payload } = action;
  switch (type) {
    case TOP_ALBUMS_RETRIEVED:
      let albumsSaved;
      if (localStorage.getItem('albums')) {
        albumsSaved = JSON.parse(localStorage.getItem('albums'));
        return [...albumsSaved];
      }
      localStorage.setItem('albums', JSON.stringify(payload));
      return [...payload];
    case ALBUM_IMAGE_RETRIEVED:
      const albumsWithImages = state.map((album) => {
        if (album.id !== payload.albumId) {
          return album;
        }
        return { ...album, imageUrl: payload.albumImage };
      });
      localStorage.setItem('albums', JSON.stringify(albumsWithImages));
      return [...albumsWithImages];

    default:
      return state;
  }
};
export const getTopAlbumsActionCreator = (topAlbums) => ({
  type: TOP_ALBUMS_RETRIEVED,
  payload: topAlbums,
});
export const getAlbumImageActionCreator = (albumImage, albumId) => ({
  type: ALBUM_IMAGE_RETRIEVED,
  payload: {
    albumImage,
    albumId,
  },
});
export const getAlbumImage = (albumId) => async (dispatch) => {
  try {
    const imagesRes = await AlbumService.getTopAlbumsImages(albumId);
    const albumImage = imagesRes.data.images[0].url;
    dispatch(getAlbumImageActionCreator(albumImage, albumId));
    return Promise.resolve(imagesRes);
  } catch (err) {
    return Promise.reject(err);
  }
};
export const getTopAlbums = () => async (dispatch) => {
  try {
    const res = await AlbumService.getTopAlbumsInfo();
    const topAlbums = [];

    res.data.albums.forEach(async (element) => {
      topAlbums.push({
        id: element.id,
        name: element.name,
        trackCount: element.trackCount,

      });
    });
    dispatch(getTopAlbumsActionCreator(topAlbums));
    return Promise.resolve(res);
  } catch (err) {
    return Promise.reject(err);
  }
};

export default reducer;
