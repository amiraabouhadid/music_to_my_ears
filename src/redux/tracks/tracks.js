import TrackService from '../../services/TrackService';

const TRACKS_RETRIEVED = 'app/tracks/TRACKS_RETRIEVED ';

const reducer = (state = [], action) => {
  const { type, payload } = action;
  switch (type) {
    case TRACKS_RETRIEVED:
      return [...payload];

    default:
      return state;
  }
};
export const getAlbumTracksActionCreator = (albumTracks) => ({
  type: TRACKS_RETRIEVED,
  payload: albumTracks,
});

export const getAlbumTracks = (albumId) => async (dispatch) => {
  try {
    const res = await TrackService.getAlbumTracks(albumId);
    const albumTracks = [];
    res.data.tracks.forEach((element) => {
      albumTracks.push({
        id: element.id,
        name: element.name,
        playbackSeconds: element.playbackSeconds,
        previewUrl: element.previewURL,
      });
    });

    dispatch(getAlbumTracksActionCreator(albumTracks));
    return Promise.resolve(res);
  } catch (err) {
    return Promise.reject(err);
  }
};
export default reducer;
