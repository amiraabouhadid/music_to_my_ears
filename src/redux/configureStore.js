import { combineReducers, createStore, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import albums from './albums/albums';

const rootReducer = combineReducers({
  albums,

});

const middlewares = [thunk];

if (process.env.NODE_ENV === 'development') {
  middlewares.push(logger);
}

const store = createStore(
  rootReducer,
  applyMiddleware(...middlewares),
);

export default store;
