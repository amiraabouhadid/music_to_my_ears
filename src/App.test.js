// import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
// import userEvent from '@testing-library/user-event';
import {
  BrowserRouter as Router,
} from 'react-router-dom';
import { Provider } from 'react-redux';
import { combineReducers, createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import albums, {
  albumsLoading, cleanUpActionCreator, getAlbumsActionCreator,
} from './redux/albums/albums';
import tracks, {
  tracksLoading, tracksCleanUpActionCreator, getAlbumTracksActionCreator,
} from './redux/tracks/tracks';
import Welcome from './components/Welcome';
import Home from './components/Home';
import TracksHeader from './components/TracksHeader';
import TracksList from './components/TracksList';

const rootReducer = combineReducers({
  albums,
  tracks,
});

const Store = createStore(
  rootReducer,
  applyMiddleware(thunk),
);
describe('Welcome', () => {
  test('welcome page renders correctly', () => {
    const welcomePage = renderer.create(
      <Provider store={Store}>
        <Router>
          <Welcome />
        </Router>

      </Provider>,
    ).toJSON();
    expect(welcomePage).toMatchSnapshot();
  });
});
describe('Home', () => {
  test('home page renders correctly', () => {
    const homePage = renderer.create(
      <Provider store={Store}>
        <Router>
          <Home />
        </Router>
      </Provider>,
    ).toJSON();
    expect(homePage).toMatchSnapshot();
  });
});
describe('Tracks ', () => {
  test('tracks page header renders correctly', () => {
    const albumProp = {
      id: 1, name: 'album1', trackCount: 2, imageUrl: 'imgURL',
    };
    const tracksHeader = renderer.create(
      <Provider store={Store}>
        <Router>
          <TracksHeader album={albumProp} />
        </Router>
      </Provider>,
    ).toJSON();
    expect(tracksHeader).toMatchSnapshot();
  });
  test('tracks list renders correctly', () => {
    const tracksProp = [{
      id: 1, name: 'track1', playbackSeconds: 200, previewUrl: 'previewURL',
    }];
    const tracksList = renderer.create(
      <Provider store={Store}>
        <Router>
          <TracksList tracks={tracksProp} loadingStatus="idle" />
        </Router>
      </Provider>,
    ).toJSON();

    expect(tracksList).toMatchSnapshot();
  });
});

describe('Albums Reducer', () => {
  test('should return the initial state', () => {
    expect(albums(undefined, {})).toEqual({
      status: 'idle',
      entities: [],
    });
  });
  test('should change status to loading when albums are being loaded', () => {
    const previousState = {
      status: 'idle',
      entities: [],
    };
    expect(albums(previousState, albumsLoading())).toEqual({
      status: 'loading',
      entities: [],
    });
  });
  test('should add retrieved albums to entities in state and change status back to idle', () => {
    const previousState = {
      status: 'loading',
      entities: [],
    };
    const albumsRetrieved = [{
      id: 1, name: 'album1', trackCount: 2, imageUrl: 'imgURL',
    }];
    expect(albums(previousState, getAlbumsActionCreator(albumsRetrieved))).toEqual({
      status: 'idle',
      entities: [{
        id: 1, name: 'album1', trackCount: 2, imageUrl: 'imgURL',
      }],
    });
  });
  test('should remove albums', () => {
    const previousState = {
      status: 'idle',
      entities: [{
        id: 1, name: 'album1', trackCount: 2, imageUrl: 'imgURL',
      }],
    };

    expect(albums(previousState, cleanUpActionCreator())).toEqual({
      status: 'idle',
      entities: [],
    });
  });
});

describe('Tracks Reducer', () => {
  test('should return the initial state', () => {
    expect(tracks(undefined, {})).toEqual({
      status: 'idle',
      entities: [],
    });
  });
  test('should change status to loading when tracks are being loaded', () => {
    const previousState = {
      status: 'idle',
      entities: [],
    };
    expect(tracks(previousState, tracksLoading())).toEqual({
      status: 'loading',
      entities: [],
    });
  });
  test('should add retrieved tracks to entities in state and change status back to idle', () => {
    const previousState = {
      status: 'loading',
      entities: [],
    };
    const trackssRetrieved = [{
      id: 1, name: 'track1', playbackSeconds: 200, previewUrl: 'previewURL',
    }];
    expect(tracks(previousState, getAlbumTracksActionCreator(trackssRetrieved))).toEqual({
      status: 'idle',
      entities: [{
        id: 1, name: 'track1', playbackSeconds: 200, previewUrl: 'previewURL',
      }],
    });
  });
  test('should remove tracks', () => {
    const previousState = {
      status: 'idle',
      entities: [{
        id: 1, name: 'track1', playbackSeconds: 200, previewUrl: 'previewURL',
      }],
    };

    expect(tracks(previousState, tracksCleanUpActionCreator())).toEqual({
      status: 'idle',
      entities: [],
    });
  });
});
