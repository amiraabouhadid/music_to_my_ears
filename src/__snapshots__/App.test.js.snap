// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home home page renders correctly 1`] = `
Array [
  <div
    className="container-fluid"
  >
    <div
      className="row header-row pt-2 text-white"
    >
      <div
        className="col-4"
      >
        <a
          className="text-light"
          href="/music_to_my_ears/"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.224 1.553a.5.5 0 0 1 .223.67L6.56 8l2.888 5.776a.5.5 0 1 1-.894.448l-3-6a.5.5 0 0 1 0-.448l3-6a.5.5 0 0 1 .67-.223z"
              fillRule="evenodd"
            />
          </svg>
        </a>
      </div>
      <div
        className="col-4 text-center"
      >
        <p />
      </div>
      <div
        className="col-4 "
      >
        <input
          className="w-75 border-0 border-rounded float-end"
          id="search-input"
          onChange={[Function]}
          placeholder="search..."
          style={
            Object {
              "display": "hidden",
            }
          }
          type="search"
          value=""
        />
      </div>
    </div>
  </div>,
  <div
    className="conatiner-fluid pink-bg"
  >
    <div
      className="row g-0"
    >
      <div
        className="col-6"
      />
      <div
        className="col-6"
      />
    </div>
  </div>,
]
`;

exports[`Tracks  tracks list renders correctly 1`] = `
<div
  className="p-2"
>
  <a
    className="text-light pt-3 d-flex flex-row flex-nowrap align-items-baseline justify-content-between"
    href="previewURL"
    style={
      Object {
        "textDecoration": "none",
      }
    }
  >
    <p
      className="fw-bold"
      style={
        Object {
          "fontSize": "0.85rem",
        }
      }
    >
      TRACK1
    </p>
    <p>
      200 seconds
      <svg
        className="mx-2 text-light"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          fillRule="evenodd"
        />
      </svg>
    </p>
  </a>
</div>
`;

exports[`Tracks  tracks page header renders correctly 1`] = `
<div
  className="p-3 d-flex flex-row align-items-baseline justify-content-between"
>
  <div
    className=""
  >
    <img
      alt="album1"
      src="imgURL"
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </div>
  <div
    className="text-white text-end"
  >
    <h2
      className=""
    >
      album1
    </h2>
    <small>
      2 tracks
    </small>
  </div>
</div>
`;

exports[`Welcome welcome page renders correctly 1`] = `
<div
  className="welcome-page text-light"
>
  <div
    className="p-4 dark-bg text-center"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M470.38 1.51L150.41 96A32 32 0 0 0 128 126.51v261.41A139 139 0 0 0 96 384c-53 0-96 28.66-96 64s43 64 96 64 96-28.66 96-64V214.32l256-75v184.61a138.4 138.4 0 0 0-32-3.93c-53 0-96 28.66-96 64s43 64 96 64 96-28.65 96-64V32a32 32 0 0 0-41.62-30.49z"
      />
    </svg>
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M138.2 22.71c-25.5-.14-45.82 38.57-50.14 80.29H32v18h55.23c.07 6.1.49 12.1 1.33 18l17.34-18h25.5c-3.8 3.9-7.9 7.7-12.1 11.8l-31.24 32.5v-.1c-.53.6-1.1 1.2-1.64 1.8H32v18h38.17c-12.95 14.6-25.91 30.2-35.67 46H34v.8c-8.09 13.3-13.89 26.8-15.44 39.9-3.5 33.5 9.52 67.3 33.39 87.3H32v18h65.54c2.56.1 5.16.1 7.86 0h20.1c3.7 20.7 7 41.2 8.9 60 .4 3.6-6.6 10.6-10 11.4 0-22-17.8-39.9-39.84-39.9-21.9 0-39.8 17.9-39.8 39.9.1 20.1 15.2 35.8 35 39.5 41.84 7.7 78.34-17.9 72.44-53-2.9-19.1-6.2-38.5-9.7-57.9H480v-18H139.3c-2.9-15.5-5.8-30.9-8.5-46h37.3c-2.6 7.8-8.1 15.9-15.4 23l3.7 20.8c15.5-11.3 31.3-26.6 35.7-43.8H480v-18H192.9c-1.6-14.5-7.8-30.8-17.8-39.9-2.5-2.3-5.1-4.3-7.8-6.1H480v-18H116.8c-1.2-7.7-1.9-11.8-2.7-16.4l-15.54 15.1c.08.4.17.9.25 1.3H72.63c12.31-16.2 26.52-30.9 41.17-46H480v-18H130.9c10.5-11.3 20.9-23.2 30.7-36.4 2-2.8 3.8-6 5.3-9.6H480v-18H172.1c2.5-12.89 2.7-26.36 1.2-35.24-2.7-16.12-14.2-42.89-31.9-44.83-1.1-.14-2.1-.22-3.2-.22zm-5.5 37c6.9.1 14.4 5.8 15.6 15.95 1.3 11.18-.3 19.87-3.8 27.34h-36c3.4-20 11.2-42.23 24.2-43.29zM60.23 249h31.24c-13.45 12.5-19.59 29.6-18.93 46H42.79c.04-5.6.98-11.3 2.97-17 4.02-10.4 8.93-20 14.47-29zm63.57 11.3c11.8-.3 26.9 5.8 34 12.9 5.7 6.2 9.6 14.1 11.2 21.8h-41.5c-2.1-11.8-4.2-23.4-6-34.6.7-.1 1.5-.1 2.3-.1zm-19.4 6.5c1.5 8.9 3.3 18.4 5.2 28.2H90.63c.1-11 4.22-21.2 13.77-28.2zM45.75 313h30.39c2.69 7.2 6.83 13.9 12.42 19.4 5.1 5.2 15.14-2.7 10-8.6-1.93-3.5-3.54-7.1-4.81-10.8h19.35c3 15 6.1 30.5 9.1 46h-20v-.1c-25.44-3.3-48.49-22.3-56.45-45.9z"
      />
    </svg>
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M98.05 18.54c-11.46-.08-23.59 1.28-36.08 3.99L130.1 261.1c-14.2-5.1-31.18-6.2-48.09-3.2-39.17 6.9-67.15 33.8-62.52 59.8 4.64 26.1 40.14 41.7 79.33 34.7 39.08-6.9 67.08-33.7 62.38-59.8-22.1-80.8-44.4-163-66.39-244.21 50.69 3.59 72.59 31.58 90.59 60.31-.5-62.33-37.7-89.81-87.35-90.16zm310.65 30.7c-13.9.1-28.8 2.59-44.1 7.22l87.1 232.44c-14.6-3.9-31.6-3.7-48.3.7-38.4 10-64.2 39-57.5 64.6 6.7 25.7 43.4 38.3 81.8 28.2 38.5-10 64.2-39 57.5-64.6-28.6-78.8-57.3-158.9-85.8-238.2 50.8-.5 74.9 25.7 95.2 52.9-5.2-58.55-40.4-83.61-85.9-83.26zM258.4 163.5l1.8 248.1c-12.3-8.6-28.4-14.3-45.6-15.9-39.4-3.8-73.7 14.5-76.2 41-2.5 26.3 27.7 50.8 67.1 54.5 39.7 3.9 73.7-14.5 76.3-40.9.2-83.8.8-168.9 1.3-253.1 47.9 17 61.6 49.8 71.2 82.4 19.9-74.1-27.9-112.3-95.9-116.1z"
      />
    </svg>
    <h1>
      MUSIC TO MY EARS
    </h1>
  </div>
  <div
    className="p-5 text-center"
  >
    <div>
      <h2>
        PICK A CATEGORY!
      </h2>
    </div>
    <div
      className=" border my-4 p-5"
      style={
        Object {
          "backgroundColor": "rgb(35,181,78)",
        }
      }
    >
      <h3>
        <a
          className="text-light"
          href="/music_to_my_ears/albums/top"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
          TOP ALBUMS
        </a>
      </h3>
    </div>
    <div
      className=" border my-4 p-5"
      style={
        Object {
          "backgroundColor": "rgb(247,90,46)",
        }
      }
    >
      <h3>
        <a
          className="text-light"
          href="/music_to_my_ears/albums/new"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
          NEW ALBUMS
        </a>
      </h3>
    </div>
    <div
      className=" border my-4 p-5"
      style={
        Object {
          "backgroundColor": "rgb(93, 93, 235)",
        }
      }
    >
      <h3>
        <a
          className="text-light"
          href="/music_to_my_ears/albums/picks"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
          STAFF PICKS
        </a>
      </h3>
    </div>
  </div>
  <div
    className="dark-bg fixed-bottom p-3 text-center"
  >
    Amira Abouhadid 
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 1024 1024"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm5.4 670c-110 0-173.4-73.2-173.4-194.9v-52.3C344 364.2 407.4 290 517.3 290c94.3 0 162.7 60.7 162.7 147.4 0 2.6-2.1 4.7-4.7 4.7h-56.7c-4.2 0-7.6-3.2-8-7.4-4-49.5-40-83.4-93-83.4-65.3 0-102.1 48.5-102.1 135.5v52.6c0 85.7 36.9 133.6 102.1 133.6 52.8 0 88.7-31.7 93-77.8.4-4.1 3.8-7.3 8-7.3h56.8c2.6 0 4.7 2.1 4.7 4.7 0 82.6-68.7 141.4-162.7 141.4z"
      />
    </svg>
     Creative Commons
  </div>
</div>
`;
